frontend/packages/kubevirt-plugin/src/components/modals/menu-actions-modals/delete-vmi-modal.tsx

```diff
@@ -0,0 +1,111 @@
+import * as React from 'react';
+import { apiVersionForModel } from '@console/internal/module/k8s';
+import { HandlePromiseProps, withHandlePromise } from '@console/internal/components/utils';
+import { YellowExclamationTriangleIcon } from '@console/shared/src/components/status/icons';
+import { getName, getNamespace } from '@console/shared/src/selectors/common';
+import {
+  createModalLauncher,
+  ModalTitle,
+  ModalBody,
+  ModalSubmitFooter,
+  ModalComponentProps,
+} from '@console/internal/components/factory';
+import { getVMIVolumes } from '../../../selectors/vmi';
+import { VMIKind } from '../../../types';
+import { deleteVMI } from '../../../k8s/requests/vmi/actions';
+import { useOwnedVolumeReferencedResources } from '../../../hooks/use-owned-volume-referenced-resources';
+import { useUpToDateVMLikeEntity } from '../../../hooks/use-vm-like-entity';
+import { VirtualMachineInstanceModel } from '../../../models';
+import { redirectToList } from './utils';
+import { VMIUsersAlert } from './vmi-users-alert';
+
+export const DeleteVMIModal = withHandlePromise((props: DeleteVMIProps) => {
+  const { inProgress, errorMessage, handlePromise, close, cancel, vmi } = props;
+
+  const vmiUpToDate = useUpToDateVMLikeEntity<VMIKind>(vmi);
+  const [deleteDisks, setDeleteDisks] = React.useState<boolean>(true);
+
+  const namespace = getNamespace(vmiUpToDate);
+  const name = getName(vmiUpToDate);
+
+  const vmiReference = {
+    name,
+    kind: VirtualMachineInstanceModel.kind,
+    apiVersion: apiVersionForModel(VirtualMachineInstanceModel),
+  } as any;
+
+  const [ownedVolumeResources, isOwnedVolumeResourcesLoaded] = useOwnedVolumeReferencedResources(
+    vmiReference,
+    namespace,
+    getVMIVolumes(vmiUpToDate, null),
+  );
+  const isInProgress = inProgress || !isOwnedVolumeResourcesLoaded;
+  const numOfAllResources = ownedVolumeResources.length;
+
+  const submit = (e) => {
+    e.preventDefault();
+
+    const promise = deleteVMI(vmiUpToDate, {
+      ownedVolumeResources,
+      deleteOwnedVolumeResources: deleteDisks,
+    });
+
+    return handlePromise(promise)
+      .then(close)
+      .then(() => redirectToList(vmiUpToDate));
+  };
+
+  const alertHref = `/k8s/ns/${namespace}/virtualmachineinstances/${name}/details#logged-in-users`;
+
+  return (
+    <form onSubmit={submit} className="modal-content">
+      <ModalTitle>
+        <YellowExclamationTriangleIcon className="co-icon-space-r" /> Delete Virtual Machine
+        Instance?
+      </ModalTitle>
+      <ModalBody>
+        Are you sure you want to delete <strong className="co-break-word">{name}</strong> in
+        namespace <strong>{namespace}</strong>?
+        {numOfAllResources > 0 && (
+          <p>
+            The following resources will be deleted along with this virtual machine instance.
```

Comment:
If I understand this correctly, this message is wrong. It looks like this is displaying the number of disks and giving a single checkbox to indicate consent to delete them or not. This message implies that there should be a list containing each of the individual resources with their own checkbox, where the user can pick which resources to delete and which to leave.